content/blog/2023/09/03/2023-09-03-jsr305-probe-blog.adoc

```diff
@@ -0,0 +1,39 @@
+---
+layout: post
+title: JSR-305 Probe Blog
+tags:
+- gsoc
+- gsoc2023
+- healthscore
+- probes
+- plugin
+authors:
+- jagruti
+opengraph:
+  image: /images/gsoc/2023/jagruti/jsr-305-probe-image.png
+sig:
+- gsoc
+---
+
+== Introduction
+
+`@Nonnull` and `@CheckForNull` annotations (part of JSR-305) have been link:https://www.jenkins.io/doc/developer/tutorial-improve/replace-jsr-305-annotations/[deprecated since 2016]. For security reasons it is important that the outdated annotations are highlighted and replaced with `javax.annotation.Nonnull` and `edu.umd.cs.findbugs.annotations.NonNull`
```

Comment:
"For security reasons" -- that sounds misleading. The real reasons are mentioned in https://groups.google.com/g/jenkinsci-dev/c/uE1wwtVi1W0/m/gLxdEJmlBQAJ.
Also `javax.annotation.Nonnull` is one of the outdated annotations, nothing should be replaced with it.

Reply:
I have added more information. Thanks for reviewing.